### PR TITLE
filter: fix a bug about ignore the DDL mistakenly

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 

This bug is found by `TICASE_3587`

recurrent:

config:
```

[filter]

rules = ['TICASE_3587.C22']

```

1. create changefeed with the config
1. run `CREATE SCHEMA TICASE_3587;` in upstream db
1. run `CREATE TABLE TICASE_3587.C22 (a int primary key);` in upstream db
1. run `insert into TICASE_3587.C22 values (1)` in upstream db

error reported in log:
```
[2020/08/17 03:23:33.498 +00:00] [DEBUG] [changefeed.go:658] ["DDL processed to make special features mysql-compatible"] [query="create database if not exists `TICASE_3587`"]
[2020/08/17 03:23:33.498 +00:00] [INFO] [mysql.go:116] ["DDL event ignored"] [query="create database if not exists `TICASE_3587`"] [startTs=418810327479615491] [commitTs=418810327492722690]
[2020/08/17 03:23:33.498 +00:00] [INFO] [changefeed.go:678] ["Execute DDL ignored"] [changefeed=2d4ee75c-6fa8-41da-96cc-565781a37037] [ddlJob="{\"id\":360,\"type\":1,\"schema_id\":359,\"table_id\":0,\"schema_name\":\"TICASE_3587\",\"state\":4,\"err\":null,\"err_count\":0,\"row_count\":0,\"raw_args\":[{\"id\":359,\"db_name\":{\"O\":\"TICASE_3587\",\"L\":\"ticase_3587\"},\"charset\":\"utf8mb4\",\"collate\":\"utf8mb4_bin\",\"state\":5}],\"schema_state\":5,\"snapshot_ver\":0,\"start_ts\":418810327479615491,\"dependency_id\":0,\"query\":\"create database if not exists `TICASE_3587`\",\"binlog\":{\"SchemaVersion\":210,\"DBInfo\":{\"id\":359,\"db_name\":{\"O\":\"TICASE_3587\",\"L\":\"ticase_3587\"},\"charset\":\"utf8mb4\",\"collate\":\"utf8mb4_bin\",\"state\":5},\"TableInfo\":null,\"FinishedTS\":418810327492722690},\"version\":1,\"reorg_meta\":null,\"priority\":0}"]

......

[2020/08/17 03:23:00.974 +00:00] [ERROR] [processor.go:1048] ["error on running processor"] [captureid=b11fafb1-8c80-483b-9922-ce2fd9240d1c] [captureaddr=ticdc1-ebvgezwk-ticdc-0.ticdc1-ebvgezwk-ticdc-peer.test-cdc.svc:8301] [changefeedid=41966440-9aa8-4855-ac2b-5fd5621cf7e6] [processorid=3dbeeb3f-086f-4bf1-9e09-a14838286c09] [error="Error 1146: Table 'TICASE_3587.C22' doesn't exist"] [errorVerbose="Error 1146: Table 'TICASE_3587.C22' doesn't exist\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20190809092503-95897b64e011/juju_adaptor.go:15\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).execDMLWithMaxRetries.func2.3\n\tgithub.com/pingcap/ticdc/cdc/sink/mysql.go:668\ngithub.com/pingcap/ticdc/cdc/sink.(*Statistics).RecordBatchExecution\n\tgithub.com/pingcap/ticdc/cdc/sink/statistics.go:86\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).execDMLWithMaxRetries.func2\n\tgithub.com/pingcap/ticdc/cdc/sink/mysql.go:659\ngithub.com/pingcap/ticdc/pkg/retry.Run.func1\n\tgithub.com/pingcap/ticdc/pkg/retry/retry.go:31\ngithub.com/cenkalti/backoff.RetryNotify\n\tgithub.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:37\ngithub.com/cenkalti/backoff.Retry\n\tgithub.com/cenkalti/backoff@v2.2.1+incompatible/retry.go:24\ngithub.com/pingcap/ticdc/pkg/retry.Run\n\tgithub.com/pingcap/ticdc/pkg/retry/retry.go:30\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).execDMLWithMaxRetries\n\tgithub.com/pingcap/ticdc/cdc/sink/mysql.go:651\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).execDMLs\n\tgithub.com/pingcap/ticdc/cdc/sink/mysql.go:779\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSinkWorker).run.func2\n\tgithub.com/pingcap/ticdc/cdc/sink/mysql.go:594\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSinkWorker).run\n\tgithub.com/pingcap/ticdc/cdc/sink/mysql.go:624\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).createSinkWorkers.func1\n\tgithub.com/pingcap/ticdc/cdc/sink/mysql.go:444\nruntime.goexit\n\truntime/asm_amd64.s:1373"]

```

The TiCDC ignore the create scheam DDL by mistake

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: pkg/filter/filter.go [hunk 1]
```diff
@@ -87,8 +87,16 @@ func (f *Filter) ShouldIgnoreDMLEvent(ts uint64, schema, table string) bool {
 
 // ShouldIgnoreDDLEvent removes DDLs that's not wanted by this change feed.
 // CDC only supports filtering by database/table now.
-func (f *Filter) ShouldIgnoreDDLEvent(ts uint64, schema, table string) bool {
-	return f.shouldIgnoreStartTs(ts) || f.ShouldIgnoreTable(schema, table)
+func (f *Filter) ShouldIgnoreDDLEvent(ts uint64, ddlType model.ActionType, schema, table string) bool {
+	var shouldIgnoreTableOrSchema bool
+	switch ddlType {
```
Comment (amyangfei) [inline pkg/filter/filter.go:92]: what should we do with `RenameTable`

Comment (leoppro) [reply pkg/filter/filter.go:92]: just skip the DDL and DMLs after this. what do you think about it?

Comment (amyangfei) [reply pkg/filter/filter.go:92]: case 1:
- rules = ['test.t1'], user create table `test`.`t1` in upstream, `test`.`t1` is replicated.
- rename `t1` to `t2`, the DDL should not be ignored
case 2:
- rules = ['test.t1'], user create table `test`.`t2` in upstream, no table is created in downstream
- rename `t2` to `t1`, DDL executed with error but ignored, the following DMLs will meet error.

Comment (leoppro) [reply pkg/filter/filter.go:92]: for case 1:
we just skip the rename DDL

for case 2:
TiCDC can't handle this scenario, manual intervention is required

**Pull request merged**